components/index.d.ts

```diff
@@ -17,390 +23,927 @@ import { ImageListItemProps } from '@mui/material/ImageListItem';
+  /**
+   * @default 'regular'
+   */
+  weight?: FontWeightVariant;
+  /**
+   * When using MUI in combination with styled/emotion, we lose access
+   * to the component prop. This appears to be a limitation of Typescript;
+   * doing this is a workaround:
+   *
+   * https://github.com/mui/material-ui/issues/15695#issuecomment-1026602904
+   */
+  component?: React$1.ElementType;
 }
 declare const Typography: (props: TypographyProps) => JSX.Element;
 
-export { Banner, BannerProps, Box, BoxProps, Button, ButtonProps, Card, CardColorProperty, CardContent, CardContentProps, CardHeader, CardHeaderProps, CardProps, ConfirmationModal, ConfirmationModalProps, ConfirmationModalVariant, DatePickerField, DatePickerProps, Dialog, DialogActionProps, DialogActions, DialogContentProps, DialogContent as DialogContentText, DialogHeader, DialogHeaderProps, DialogProps, DialogTopContent, DialogTopContentProps, FontWeightValue, FontWeightVariant, IconButton, IconButtonProps, ImageGrid, ImageGridItem, ImageGridItemProps, ImageGridProps, ImageItem, InformationCard, InformationCardProps, InlineNotification, InlineNotificationProps, LineItem, LineItemProps, List, ListItem, ListItemProps, ListProps, NotificationVariant, StatefulTextField, SuggestedAction, SuggestedActionAccordion, SuggestedActionAccordionProps, SuggestedActionProps, TextField, TextFieldProps, Typography, TypographyProps, fontWeights };
+interface MenuProps extends MenuProps$1 {
+  /**
+   * pass in menu items or buttons as children
+   * @default undefined
+   */
+  children: React$1.ReactNode;
+  /**
+   * an HTML element that's used to set the position of the menu
+   * @default undefined
+   */
+  anchorEl: HTMLElement | null;
+  /**
+   * if true, menu is shown
+   * @default undefined
+   */
+  open: boolean;
+}
+declare const Menu: ({
+  children,
+  anchorEl,
+  open,
+  ...props
+}: MenuProps) => JSX.Element;
+
+interface MenuTopContentProps {
+  onClose?: (args?: any) => void;
+  title?: string;
+}
+declare const MenuTopContent: ({
+  onClose,
+  title,
+}: MenuTopContentProps) => JSX.Element;
+
+export {
+  Banner,
+  BannerProps,
+  Box,
+  BoxProps,
+  Button,
+  ButtonProps,
+  Card,
+  CardColorProperty,
+  CardContent,
+  CardContentProps,
+  CardHeader,
+  CardHeaderProps,
+  CardProps,
+  ConfirmationModal,
+  ConfirmationModalProps,
+  ConfirmationModalVariant,
+  DatePickerField,
+  DatePickerProps,
+  Dialog,
+  DialogActionProps,
+  DialogActions,
+  DialogContentProps,
+  DialogContent as DialogContentText,
+  DialogHeader,
+  DialogHeaderProps,
+  DialogProps,
+  DialogTopContent,
+  DialogTopContentProps,
+  FontWeightValue,
+  FontWeightVariant,
+  IconButton,
+  IconButtonProps,
+  ImageGrid,
+  ImageGridItem,
+  ImageGridItemProps,
+  ImageGridProps,
+  ImageItem,
+  InformationCard,
+  InformationCardProps,
+  InlineNotification,
+  InlineNotificationProps,
+  LineItem,
+  LineItemProps,
+  List,
+  ListItem,
+  ListItemProps,
+  ListProps,
+  Menu,
```

Comment:
@smurmus it's ok that menu is exported here?

Reply:
yep, since we don't want to override the MUI version to avoid impacting other files, it makes sense that it's just added to the `components` index file